xml/src/main/java/org/ehcache/xml/CoreCacheConfigurationParser.java

```diff
@@ -0,0 +1,100 @@
+
+public class CoreCacheConfigurationParser {
+
+  public <K, V> CacheConfigurationBuilder<K, V> parseConfiguration(CacheTemplate cacheDefinition, ClassLoader cacheClassLoader,
+                                                                   CacheConfigurationBuilder<K, V> cacheBuilder) throws ClassNotFoundException, IllegalAccessException, InstantiationException {
+    final Expiry parsedExpiry = cacheDefinition.expiry();
+    if (parsedExpiry != null) {
+      cacheBuilder = cacheBuilder.withExpiry(getExpiry(cacheClassLoader, parsedExpiry));
+    }
+
+    EvictionAdvisor evictionAdvisor = getInstanceOfName(cacheDefinition.evictionAdvisor(), cacheClassLoader, EvictionAdvisor.class);
+    cacheBuilder = cacheBuilder.withEvictionAdvisor(evictionAdvisor);
+
+    return cacheBuilder;
+  }
+
+  @SuppressWarnings({"unchecked", "deprecation"})
+  private static ExpiryPolicy<? super Object, ? super Object> getExpiry(ClassLoader cacheClassLoader, Expiry parsedExpiry)
+    throws ClassNotFoundException, InstantiationException, IllegalAccessException {
+    final ExpiryPolicy<? super Object, ? super Object> expiry;
+    if (parsedExpiry.isUserDef()) {
+      ExpiryPolicy<? super Object, ? super Object> tmpExpiry;
+      try {
+        tmpExpiry = getInstanceOfName(parsedExpiry.type(), cacheClassLoader, ExpiryPolicy.class);
+      } catch (ClassCastException e) {
+        tmpExpiry = ExpiryUtils.convertToExpiryPolicy(getInstanceOfName(parsedExpiry.type(), cacheClassLoader, org.ehcache.expiry.Expiry.class));
+      }
+      expiry = tmpExpiry;
+    } else if (parsedExpiry.isTTL()) {
+      expiry = ExpiryPolicyBuilder.timeToLiveExpiration(Duration.of(parsedExpiry.value(), parsedExpiry.unit()));
+    } else if (parsedExpiry.isTTI()) {
+      expiry = ExpiryPolicyBuilder.timeToIdleExpiration(Duration.of(parsedExpiry.value(), parsedExpiry.unit()));
+    } else {
+      expiry = ExpiryPolicyBuilder.noExpiration();
+    }
+    return expiry;
+  }
+
+  public CacheType unparseConfiguration(CacheConfiguration<?, ?> cacheConfiguration, CacheType cacheType) {
+    ExpiryPolicy<?, ?> expiryPolicy = cacheConfiguration.getExpiryPolicy();
+    if (expiryPolicy != null) {
+      ExpiryType expiryType = new ExpiryType();
+      if (expiryPolicy == ExpiryPolicy.NO_EXPIRY) {
+        expiryType.withNone(new ExpiryType.None());
+      } else if (expiryPolicy instanceof ExpiryPolicyBuilder.TimeToLiveExpiryPolicy) {
+        Duration ttl = expiryPolicy.getExpiryForCreation(null, null);
+        expiryType.withTtl(new TimeType().withValue(BigInteger.valueOf(ttl.getSeconds())).withUnit(TimeUnit.SECONDS));
+      } else if (expiryPolicy instanceof ExpiryPolicyBuilder.TimeToIdleExpiryPolicy) {
+        Duration tti = expiryPolicy.getExpiryForCreation(null, null);
+        expiryType.withTti(new TimeType().withValue(BigInteger.valueOf(tti.getSeconds())).withUnit(TimeUnit.SECONDS));
+      } else {
+        expiryType.setClazz(expiryPolicy.getClass().getName());
```

Comment:
This isn't generally true right?  We don't know how this class works and therefore don't know if it's safe to instantiate in this way?

Reply:
True. Missed that. But then this will get tricky. Our current API does not support class based instantiation of either `ExpiryPolicy` or the `EvictionAdvisor`. I'm wondering how we can determine if those can be safely instantiated when unparsed back. If not, we might have to say that we don't support translation of custom class based expiry and eviction.

Reply:
For now it means that... no way of knowing where the instance came from.  I don't think this will be a huge loss.